integrations/nginx/install.go

```diff
@@ -0,0 +1,349 @@
+package nginx
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/akitasoftware/go-utils/optionals"
+	"github.com/pkg/errors"
+)
+
+/* Top-level error to report to the user. */
+type InstallationError struct {
+	Wrapped error
+	Remedy  string
+}
+
+func (e *InstallationError) Unwrap() error {
+	return e.Wrapped
+}
+
+func (e *InstallationError) Error() string {
+	return fmt.Sprintf("Couldn't automatically install the Akita NGX module: %v", e.Wrapped)
+}
+
+var _ error = (*InstallationError)(nil)
+
+var (
+	unsupportedError = &InstallationError{
+		Wrapped: errors.New("The version of NGINX that is installed does not have a precompiled module."),
+		Remedy:  "Please try building the module from source, using the directions at https://github.com/akitasoftware/akita-nginx-module#building-the-module-from-source",
+	}
+
+	uninstalledError = &InstallationError{
+		Wrapped: errors.New("NGINX is not installed, or 'nginx' is not in the current path"),
+		Remedy:  "Please run this command on the machine where NGINX is installed.",
+	}
+
+	downloadError = &InstallationError{
+		Wrapped: errors.New("The precompiled module could not be downloaded from Akita."),
+		Remedy:  "Please download the module from https://github.com/akitasoftware/akita-nginx-module/releases/latest and complete the installation by copying it into the NGINX module directory.",
+	}
+
+	symlinkError = &InstallationError{
+		Wrapped: errors.New("The precompiled module was installed, but a symbolic link to it could not be created."),
+		Remedy:  "Please create a symbolic link in the module directory from ngx_http_akita_module.so to the installed module.",
+	}
+)
+
+func NewCopyError(path string) error {
+	return &InstallationError{
+		Wrapped: errors.New("The precompiled module could not be installed into the NGINX directory."),
+		Remedy:  fmt.Sprintf("Please complete the installation by copying %s into the NGINX module directory.\n", path),
+	}
+}
+
+type InstallArgs struct {
+	DryRun bool
+
+	// Destination directory for the module
+	DestDir optionals.Optional[string]
+}
+
+// Install the precompiled NGINX module from the latest release on GitHub.
+// If we can't find the Nginx version or platform,
+// Return an error to show to the user.
+func InstallModule(args *InstallArgs) error {
```

Comment:
Does this always return an `InstallationError`? If yes, let's change the return type.

Reply:
It can return other sorts of errors currently, but maybe it shouldn't.